unidist/core/backends/mpi/core/communication.py

```diff
@@ -371,7 +464,7 @@ def mpi_isend_buffer(comm, buffer_size, buffer, dest_rank):
     return requests
 
 
-def mpi_recv_buffer(comm, source_rank):
+def mpi_recv_buffer(comm, source_rank, result_buffer=None):
```

Comment:
We don't have to add this parameter now because we always receive buf size so let's revert the change.

Reply:
No, when we send a shared buffer to another host for synchronization, we haven't to get the size and create the receive buffer separately.

Reply:
to be discussed offline

Reply:
Let's align this function with mpi_send_buffer, i.e., replace `result_buffer` to `recv_size` and correct the behavior of this function.

Reply:
It is impossible for this case. We need to receive data to a pre-allocated shared memory, and not create it. Or use a separate method as I suggested earlier.